dataprofiler/profilers/profile_builder.py

```diff
@@ -2240,7 +2240,20 @@ def _update_null_replication_metrics(self, clean_samples: Dict) -> None:
         :param clean_samples: input cleaned dataset
         :type clean_samples: dict
         """
-        data = pd.DataFrame(clean_samples).apply(pd.to_numeric, errors="coerce")
+        data = pd.DataFrame(clean_samples)
+
+        # If the last row is all null, then add rows to the data DataFrame
+        max_null_index = max(
+            [max(i) for i in getattr(self._profile[0], "null_types_index").values()],
+            default=0,
+        )
+        if max_null_index > data.index.max():
+            data.loc[max_null_index] = {}
+
+        # Fill in missing rows with NaN and convert types to numeric
+        data = data.reindex(range(data.index.max() + 1), fill_value=np.nan).apply(
```

Comment:
this could be really expensive for large data. Is there another way around this?

Reply:
What specific error is requiring this change / where?

Reply:
+1 good catch @JGSweets

Reply:
If a row is all null, the dataframe will be missing a row, which causes an error on the line
`sum_null = data.iloc[null_indices, data.columns != col_id].sum().to_numpy()`.
`reindex` fills in these missing rows with `nan`.

Reply:
The `.reindex()` though when done on an entire dataset could be very costly in terms of runtime; so that is the concern around this operation that is being added to the process.

Reply:
If a row is null, could we do a join with non-null rows? I believe we have the info on what rows are completely null actually.

Reply:
`null_indices + non-complete null rows`?

Reply:
or even better potentially would be `null_indices - complete null rows` (theoretically there should be less completely null rows)

Reply:
Another way would be to reindex a single column in `clean_samples` and then combine all of them into `data`, which should also fix the bug.

Reply:
That would work too

Reply:
why can't we do it w/o re-indexing?